libraries/chain/asset_evaluator.cpp

```diff
@@ -391,27 +391,61 @@ void check_children_of_bitasset(database& d, const asset_update_bitasset_operati
 
             FC_ASSERT( !new_backing_asset.is_market_issued(),
                   "A non-blockchain controlled BitAsset would be invalidated by changing this backing asset.");
+
+            // Check if the new backing asset is itself backed by something. It must be CORE or a UIA
+            if ( new_backing_asset.is_market_issued() )
+            {
+               const asset_object& backing_backing_asset = new_backing_asset.bitasset_data(d).asset_id(d);
+               if ( backing_backing_asset.get_id() != asset_id_type() )
+               {
+                  FC_ASSERT( !backing_backing_asset.is_market_issued(), "A BitAsset cannot be backed by a BitAsset that itself "
+                        "is backed by a BitAsset.");
+               }
+            }
          }
          else
          {
             if( child.get_id() == op.new_options.short_backing_asset )
+            {
                wlog( "Before hf-922-931, modified an asset to be backed by another, but would cause a continuous "
                      "loop. A cannot be backed by B which is backed by A." );
+               return;
+            }
 
             if( child.issuer == GRAPHENE_COMMITTEE_ACCOUNT )
             {
                wlog( "before hf-922-931, modified an asset to be backed by a non-CORE, but this asset "
                         "is a backing asset for a committee-issued asset. This occurred at block ${b}",
                         ("b", d.head_block_num()));
+               return;
             }
             else
```

Comment:
Due to the `return` above, this `else` can be removed.

Reply:
I believe it better to remove ``return``. Spurious exits make logic more difficult to follow.